### ISSUE TITLE
KFAC Norm Constraint

### ISSUE THREAD

**Issue by Baiyu-Su** [google-deepmind/kfac-jax#160]
Hi,

In the documentation of applying a norm constraint to the update gradient, it says:
```
norm_constraint: Scalar. If specified, the update is scaled down so that
        its approximate squared Fisher norm ``v^T F v`` is at most the specified
        value. (Note that here ``F`` is the approximate curvature matrix, not
        the exact.)
```
and the corresponding part of the code:
```
preconditioned_grads = self.estimator.multiply_inverse(
        state=state.estimator_state,
        parameter_structured_vector=grads,
        identity_weight=self.l2_reg + damping,
        exact_power=self._use_exact_inverses,
        use_cached=self._use_cached_inverses,
        pmap_axis_name=self.pmap_axis_name,
    )
if self._norm_constraint is not None:

      assert not self._use_adaptive_learning_rate
      assert coefficient is not None

      sq_norm_grads = utils.inner_product(preconditioned_grads, grads)

      sq_norm_scaled_grads = sq_norm_grads * coefficient ** 2

      max_coefficient = jnp.sqrt(self._norm_constraint / sq_norm_scaled_grads)
      coefficient = jnp.minimum(max_coefficient, 1)

      preconditioned_grads = utils.scalar_mul(preconditioned_grads, coefficient)
```
However, as far as I am aware, the `preconditioned_grads` is F^-1 v, so the `sq_norm_grads` is actually computing v^T F^-1 v instead of v^T F v as documented. Did I understand it correctly or it is intended to be like so?

**Comment by botev** [comment]
So the norm constraint is on the preconditioned gradient, which is `v = F^-1 g`. Then `v^T F v` = g^T F^-1 g = v^T g`

**Comment by Baiyu-Su** [comment]
Got it! Many thanks!
